zproject/backends.py

```diff
@@ -282,5 +286,28 @@ def get_full_name(self, *args, **kwargs):
 
     def do_auth(self, *args, **kwargs):
         kwargs['return_data'] = {}
-        user_profile = super(GitHubAuthBackend, self).do_auth(*args, **kwargs)
+        backend = GithubOrganizationOAuth2(self.strategy, self.redirect_uri)
+        try:
+            user_profile = backend.do_auth(*args, **kwargs)
+        except AuthFailed:
+            logging.info("User profile not member of organisation.")
+            user_profile = None
```

Comment:
I think we can probably just inspect the settings for whether these settings are enabled:
* `SOCIAL_AUTH_GITHUB_ORG_NAME = ''`
* `SOCIAL_AUTH_GITHUB_TEAM_ID = ''`
And choose which backend to use based on that, rather than checking all of the backends.